apps/sms/js/activity_handler.js

```diff
@@ -250,54 +259,30 @@ var ActivityHandler = {
     });
   },
 
-  // The unsent confirmation dialog provides 2 options: edit and discard
-  // discard: clear the message user typed
-  // edit: continue to edit the unsent message and ignore the activity
-  displayUnsentConfirmation: function ah_displayUnsentConfirmtion(activity) {
-    var msgDiv = document.createElement('div');
-    msgDiv.innerHTML = '<h1 data-l10n-id="unsent-message-title"></h1>' +
-                       '<p data-l10n-id="unsent-message-description"></p>';
-    var options = new OptionMenu({
-      type: 'confirm',
-      section: msgDiv,
-      items: [{
-        l10nId: 'unsent-message-option-edit',
-        method: function editOptionMethod() {
-          // we're already in message app, we don't need to do anything
-        }
-      },
-      {
-        l10nId: 'unsent-message-option-discard',
-        method: (activity) => {
-          ThreadUI.discardDraft();
-          this.launchComposer(activity);
-        },
-        params: [activity]
-      }]
-    });
-    options.show();
-  },
-
-  // Launch the UI properly
-  launchComposer: function ah_launchComposer(activity) {
-    Navigation.toPanel('composer', { activity: activity });
-  },
-
   // Check if we want to go directly to the composer or if we
   // want to keep the previously typed text
   triggerNewMessage: function ah_triggerNewMessage(body, number, contact) {
-     var activity = {
+    var composerArgs = {
+      activity: {
         body: body || null,
         number: number || null,
         contact: contact || null
-      };
+      }
+    };
 
     if (Compose.isEmpty()) {
-      this.launchComposer(activity);
-    } else {
-      // ask user how should we do
-      ActivityHandler.displayUnsentConfirmation(activity);
+      Navigation.toPanel('composer', composerArgs);
+      return;
     }
+
+    Utils.confirm(
```

Comment:
The only thing is that we can someday extract number/email->thread_or_contact logic from ActivityHandler and use it here as well, so that when you'd like to send a message to number or email from message bubble and you already have appropriate thread - you'll navigate there instead of composer